addons/mail/static/src/core/common/link_preview_model.js

```diff
@@ -40,12 +43,31 @@ export class LinkPreview extends Record {
     }
 
     get isVideo() {
-        return Boolean(!this.isImage && this.og_type && this.og_type.startsWith("video"));
+        const fileExt = this.og_title?.split(".").pop();
+        return (
+            VIDEO_EXTENSIONS.has(fileExt) ||
+            Boolean(!this.isImage && this.og_type && this.og_type.startsWith("video"))
+        );
     }
 
     get isCard() {
         return !this.isImage && !this.isVideo;
     }
+
+    get videoURL() {
+        const youtubeVideoId = extractYouTubeVideoId(this.source_url);
+        if (youtubeVideoId) {
+            const youtubeUrl = new URL(`/embed/${youtubeVideoId}`, "https://www.youtube.com");
+            youtubeUrl.searchParams.set("autoplay", "1");
+            return youtubeUrl.toString();
+        }
+        const driveVideoId = extratGoogleDriveVideoId(this.source_url);
+        if (driveVideoId) {
+            const driveUrl = new URL(`/file/d/${driveVideoId}/preview`, "https://drive.google.com");
+            return driveUrl.toString();
+        }
+        return null;
+    }
```

Comment:
This isn't much, but could be nice a test that checks some links from youtube / gdrive are formatted for embed.

The test gives a rough idea on the simplicity of support, and also since the test exists in its own context there's less risk to break it by changing code of link preview (which likely will become ugly with support on other services and more).